packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts

```diff
@@ -115,10 +115,42 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
     const call: AwsSdkCall | undefined = event.ResourceProperties[event.RequestType];
 
     if (call) {
-      const awsService = new (AWS as any)[call.service]({
-        apiVersion: call.apiVersion,
-        region: call.region,
-      });
+      let awsService;
+
+      if (call.assumedRoleArn) {
+
+        const stsService = new (AWS as any).STS({
+          apiVersion: call.apiVersion,
+          region: call.region,
+        });
+
+        const timestamp = (new Date()).getTime();
+
+        const params = {
+          RoleArn: call.assumedRoleArn,
+          RoleSessionName: `${physicalResourceId}-${timestamp}`,
+        };
+
+        const assumeRoleResponse = await stsService.assumeRole(params).promise();
+
+        const creds = new AWS.Credentials({
+          accessKeyId: assumeRoleResponse.Credentials.AccessKeyId,
+          secretAccessKey: assumeRoleResponse.Credentials.SecretAccessKey,
+          sessionToken: assumeRoleResponse.Credentials.SessionToken,
+        });
+
+        awsService = new AWS[call.service]({
```

Comment:
Please dedupe between this and the other instantiation.

Reply:
Is this still valid with using `ChainableTemporaryCredentials`, or what does dedupe mean in this case?